adapta/storage/blob/s3_storage_client.py

```diff
@@ -31,22 +32,27 @@
 
 
 @final
-class S3StorageClient(StorageClient, ABC):
+class S3StorageClient(StorageClient):
     """
     S3 Storage Client.
     """
 
-    def __init__(self, *, base_client: AwsClient):
+    def __init__(self, *, base_client: AwsClient, endpoint_url: Optional[str] = None):
         super().__init__(base_client=base_client)
         if base_client.session is None:
             raise ValueError("AwsClient.initialize_session should be called before accessing S3StorageClient")
-        self._s3_resource = base_client.session.resource("s3")
+        endpoint_url = endpoint_url or base_client.endpoint
+        self._s3_resource = base_client.session.resource("s3", endpoint_url=endpoint_url)
 
     def get_blob_uri(self, blob_path: DataPath, **kwargs) -> str:
+        """Returns the URI for a blob in S3 storage.
+
+        :param blob_path: Path to blob
+
+        :return: The URI for the given blob path
         """
-        Not implemented in S3 Client
-        """
-        raise NotImplementedError("Not implemented in S3StorageClient")
+        s3_path = cast_path(blob_path)
+        return f"s3://{s3_path.bucket}/{s3_path.path}"
```

Comment:
This is not correct. In `StorageClient` contract, `get_blob_uri` returns a signed url for the blob that you can GET to download the blob contents